bundle/src/main/java/com/adobe/acs/tools/csv_asset_importer/impl/UnzipUtility.java

```diff
@@ -0,0 +1,103 @@
+package com.adobe.acs.tools.csv_asset_importer.impl;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * This utility extracts files and directories of a standard zip file to a
+ * destination directory.
+ *
+ */
+public class UnzipUtility {
+	/**
+	 * Size of the buffer to read/write data
+	 */
+	private static final int BUFFER_SIZE = 4096;
+
+	/**
+	 * Extracts a zip file specified by the zipFilePath to a directory specified
+	 * by destDirectory (will be created if does not exists)
+	 * 
+	 * @param zipFilePath
+	 * @param destDirectory
+	 * @throws IOException
+	 */
+	public static void unzip(String zipFilePath, String destDirectory) throws IOException {
+		File destDir = new File(destDirectory);
+		if (!destDir.exists()) {
+			destDir.mkdir();
+		}
+		ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath));
+		ZipEntry entry = zipIn.getNextEntry();
+		// iterates over entries in the zip file
+		while (entry != null) {
+			String filePath = destDirectory + File.separator + entry.getName();
+			if (!entry.isDirectory()) {
+				// if the entry is a file, extracts it
+				extractFile(zipIn, filePath);
+			} else {
+				// if the entry is a directory, make the directory
+				File dir = new File(filePath);
+				dir.mkdir();
+			}
+			zipIn.closeEntry();
+			entry = zipIn.getNextEntry();
+		}
+		zipIn.close();
+	}
+
+	/**
+	 * Extracts a zip file specified by the zipFilePath to a directory specified
+	 * by destDirectory (will be created if does not exists)
+	 * 
+	 * @param zipFilePath
+	 * @param destDirectory
+	 * @throws IOException
+	 */
+	public static void unzip(InputStream inputStream, String destDirectory) throws IOException {
+		File destDir = new File(destDirectory);
+		if (!destDir.exists()) {
+			destDir.mkdir();
+		}
+		ZipInputStream zipIn = new ZipInputStream(inputStream);
+		ZipEntry entry = zipIn.getNextEntry();
+		// iterates over entries in the zip file
+		while (entry != null) {
+			String filePath = destDirectory + File.separator + entry.getName();
+			if (!entry.isDirectory()) {
+				// if the entry is a file, extracts it
+				extractFile(zipIn, filePath);
+			} else {
+				// if the entry is a directory, make the directory
+				File dir = new File(filePath);
+				dir.mkdir();
+			}
+			zipIn.closeEntry();
+			entry = zipIn.getNextEntry();
+		}
+		zipIn.close();
+	}
+
+	/**
+	 * Extracts a zip entry (file entry)
+	 * 
+	 * @param zipIn
+	 * @param filePath
+	 * @throws IOException
+	 */
+	private static void extractFile(ZipInputStream zipIn, String filePath) throws IOException {
```

Comment:
Rather than extracting the files and importing them, which creates a lot of IO and uses a lot of temp space, why not just read the zip contents in a stream and work off the stream?  You can take a zip file input stream and pass that directly in to the asset API to create assets.  The result would be zip contents directly ingested with no middle steps.  This would greatly improve throughput and reduce resource usage.